### ISSUE TITLE
borg:  : [Errno 2] No such file or directory (filename too shortened to be useful)

### ISSUE THREAD

**Issue by tgharold** [borgbackup/borg#123]
Two issues:

1) The filename is being shortened too much to tell what is going on.  it should probably be reported on the next line so that it can be much wider.  Instead of trying to tack it onto the [Errno 2] line.  Or it needs to look at terminal width before truncating the entire message to 80 characters?

2) It looks like the single quotes are in odd places, but I'm not sure if that's due to a space in the directory name, or a single quote in the directory name.  Due to the way things are trimmed off, I can't tell.

Two copies of the error. This is under Cygwin64.
```
borg:  : [Errno 2] No such file or directory: ' 'ers/Scr...Notes - Shortcut.lnk
borg:  : [Errno 2] No such file or directory: ' 'ers/Scr...5477bf025b30f3b92ddd
```

**Comment by ThomasWaldmann** [comment]
Can't reproduce. I tried to backup a non-existing (and very long) filename on cygwin64 and it worked without cosmetic issues. it does not shorten the filename for latest git code.

**Comment by tgharold** [comment]
With --verbose

A /cygdrive/c/Users/USERNAME/AppData/Local/Google/Chrome/User Data/Profile 1/Extension Rules/CURRENT
/home/USERNAME/bin/borg-snapshots: line 9: 16860 Segmentation fault      (core dumped)

$ cat python3.4m.exe.stackdump
Exception: STATUS_ACCESS_VIOLATION at rip=00400DD1884
rax=0000000073FC18F6 rbx=0000000017ABD088 rcx=000006FFF0BD100F
rdx=0000000600322120 rsi=000000060030C580 rdi=FFFFFFFFF3BD100D
r8 =00000000000219BB r9 =00000000CFF063D9 r10=0000000000000FFF
r11=000006FFFCFDE645 r12=000000000000001F r13=000000000000FFFF
r14=0000000000000FFF r15=000006FFF0BD0010
rbp=000000000C42EFF1 rsp=000000000022B070
program=C:\cygwin64\bin\python3.4m.exe, pid 16860, thread main
cs=0033 ds=002B es=002B fs=0053 gs=002B ss=002B

**Comment by ThomasWaldmann** [comment]
@tgharold any additional input so I can reproduce?

**Comment by tgharold** [comment]
I've been running the master branch the past week or so, so I haven't seen it.  What I have seen is that filenames are being repeated in the error output line.  Possibly because I'm using "--progress" and "--stats"

borg: /cygdrive/c/Users/USERNAME/AppData/Local/ElevatedDiagnostics: [Errno 13] Permission denied: '/cygdrive/c/Users/USERNAMEAppData/Local/ElevatedDiagnostics'

So they're a bit repetitive.

**Comment by ThomasWaldmann** [comment]
the repetition comes from:
a) first borg outputs the filename (in any case)
b) borg outputs the string representation of the exception class

in case b) contains the filename (not sure if it does in any case), you see the repetition.

**Comment by ThomasWaldmann** [comment]
I think this is just a case of an error msg partially overwriting the progress indication line.
So, the filename in the error msg was `' '`. What you see to the right of that was just the progress line that was output right before the error happened.

**Comment by ThomasWaldmann** [comment]
an easy fix could be to just print a \n if the progress indication was active before printing any other "end of program" (traceback, rc code, ...) msg.

**Comment by ThomasWaldmann** [comment]
Closing this, will be solved by UI refactoring #314.
